llvm/lib/CodeGen/ExpandFp.cpp

```diff
@@ -941,10 +941,9 @@ static void scalarize(Instruction *I,
       llvm_unreachable("Unsupported instruction type");
 
     Result = Builder.CreateInsertElement(Result, NewOp, Idx);
-    if (auto *ScalarizedI = dyn_cast<Instruction>(NewOp)) {
-      ScalarizedI->copyIRFlags(I, true);
-      Worklist.push_back(ScalarizedI);
-    }
+    Instruction *ScalarizedI = cast<Instruction>(NewOp);
```

Comment:
I'm not sure removing the dyn_cast is safe, it's there to guard against the case where the IRBuilder constant folded something. I'm surprised there is no test for this case

Reply:
if there aren't tests where each and both operand are constants / constant vectors there should be